python/lsst/pipe/base/argumentParser.py

```diff
@@ -715,9 +716,10 @@ def obeyShowArgument(showOpts, config=None, exit=False):
         return
 
     for what in showOpts:
-        mat = re.search(r"^config(?:=(.+))?", what)
-        if mat:
-            pattern = mat.group(1)
+        matConfig = re.search(r"^config(?:=(?:config.)?(.+))?", what)
+        matHistory = None if matConfig else re.search(r"^history(?:=(?:config.)?(.+))?", what)
```

Comment:
I find this too confusing. The search string "^history(?:=(?:config.)?(.+))?" seems far trickier than is required, and making matHistory use the results of matConfig seems needlessly complicated. It would be far clearer to keep the two independent of each other, even if it made the code a bit more verbose.

The number of options is large enough now that it may be worth a different approach, such as first testing all options that have no associated data, then using something like this for the remaining possibilities:
```
if what == "tasks":
   ....
elif what == "data":
    ....
...  # test all items that cannot have associated data first, then...
else:
    item, data = what.split("=", 1)
    if item == "config":
        configMatch = ...
    elif item == "history":
        historyMatch = ...
```